src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -214,7 +215,31 @@ public async Task<FormulaValue> EvalAsync(string expressionText, CancellationTok
             return result;
         }
 
-        // Preview functionality 
+        internal void AddUserFunction(string script, CultureInfo locale)
+        {
+            if (UserDefinitions.ProcessUserDefinitions(script, new ParserOptions() { AllowsSideEffects = true, Culture = locale }, out var userDefinitionResult) && !userDefinitionResult.HasErrors)

```

Comment:
>ProcessUserDefinitions [](http://example.com/codeflow?start=32&length=22)

With the changes made recently, ProcessUserDefinitions will always returns true which is wrong and needs to be refactored.

ProcessUserDefinitions() can just return UserDefinitionResult instead of bool, with that change in place the above condition can be re-written

var result = UserDefinitions.ProcessUserDefinitions(script, new ParserOptions() { AllowsSideEffects = true, Culture = locale });

if (result.HasErrors){ ... }